1

zzzlou opened the pull request:
1

j-lum committed:
Merge pull request #3 from j-lum/inline-javafx-tutorials

Inline JavaFX tutorials

j-lum committed:
Update Gradle tutorial to reflect new checkstyle config

j-lum committed:
Fix typo referring to `HelloWorld`

j-lum committed:
Add images to JavaFX tutorial 3

j-lum committed:
Merge pull request #7 from j-lum/tutorial-fix

Address issues #5 and #6

j-lum committed:
Remove references to DukeStub

j-lum committed:
Merge pull request #8 from j-lum/tutorial-fix

Remove references to DukeStub

j-lum committed:
Change indentation in FXML samples to match AB-3

j-lum committed:
Merge pull request #14 from j-lum/fxml-indent

Change indentation in FXML samples to match AB-3

j-lum committed:
JavaFX tutorial: Support cross-platform JARs

The OpenJFX plugin expects applications to be modular and bundled
with jlink, resulting in fat jars that are not cross-platform. Let's
manually include the required dependencies so that shadow can package
them properly.

j-lum committed:
Merge pull request #35 from j-lum/xplatformjfx

JavaFX tutorial: Support cross-platform JARs

jiachen247 committed:
Remove unnecessary override annotation

Remove extra '@Override' at the start of the code example.

j-lum committed:
Merge pull request #41 from jiachen247/patch-2

Remove unnecessary override annotation

Commit:
Remove generic tutorials

Let's remove generic tutorials from this repo as they are not specific
to this project tempalte

Commit:
Add text-ui-test files

Commit:
Add Gradle support

Commit:
Add Gradle support

damithc committed:
README.md: Fix typo of => or

Commit:
Improve runtest.* scripts

* Improve the javac command to work for multiple java files.
* In runtest.bat, delete ACTUAL.TXT only if it exists.

Commit:
README: Update setup instructions to match the latest IDE version

Commit:
README.md: Tweak template text

* Let's replace 'Feature 1' with 'Feature-ABC'
  reason: some students kept the 'Feature 1' in the final UG.
* Let's tweak the template for features and usage
  reasons: to fix grammar, to distinguish between output and outcome

damithc committed:
README: Mention the project language level setting

seanleong339 committed:
.gitignore: Fix ACTUAL.txt -> ACTUAL.TXT  (#73)

text-ui-test/runtest.bat|sh scripts generate a file ACTUAL.TXT.
However, .gitignore uses ACTUAL.txt, which means the generated
file will not be ignored by Git on non-Windows OS.

Let's update .gitignore as ACTUAL.txt -> ACTUAL.TXT

Eclipse-Dominator committed:
Bump gradle and lib version

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
README: Update the JDK version 11 -> 17

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

zzzlou committed:
added name and greeting

zzzlou committed:
added echo

zzzlou committed:
added add and list

zzzlou committed:
added mark and unmark

zzzlou committed:
added 3 types of events

zzzlou committed:
added error handling

zzzlou committed:
added delete function

zzzlou committed:
Add save

zzzlou committed:
Merge branch 'branch-Level-7'

"necessary"

zzzlou committed:
Add date

zzzlou committed:
Merge branch 'branch-Level-8'
necessary

zzzlou committed:
Add more OOP

zzzlou committed:
Add more OOP

zzzlou committed:
Add packaging

zzzlou committed:
Add packaging

zzzlou committed:
Merge branch 'add-gradle-support'
necessary

zzzlou committed:
Add gradle test

zzzlou committed:
Add gradle test

zzzlou committed:
Add JavaDoc

zzzlou committed:
Add JavaDoc

zzzlou committed:
Add formatting

zzzlou committed:
Add formatting

zzzlou committed:
Add find

zzzlou committed:
Add find

zzzlou committed:
Merge branch 'Level-9'

zzzlou committed:
Add checkstyle

zzzlou committed:
Add checkstyle

zzzlou committed:
Add some staff

zzzlou committed:
Merge branch 'branch-A-Checkstyle'
necessary

zzzlou committed:
Add GUI

zzzlou committed:
Merge branch 'branch-Level-10'
necessary

antgyf reviewed (changes requested):
LGTM other than a few nits
- consider case if time is omitted from deadline or event, or case where deadline and time format is wrong.

yooplo commented on src/main/java/zzbot/DialogBox.java:
```diff
@@ -0,0 +1,60 @@
+    public static DialogBox getUserDialog(String text, Image img) {
+        return new DialogBox(text, img);
+    }
+
```
I would think that javadocs should be added here since it is a public method.

yooplo commented on src/main/java/zzbot/Storage.java:
```diff
@@ -0,0 +1,66 @@
+        s.close();
+
+        return arr;
+
```
extra line here haha

yooplo commented on src/main/java/zzbot/Ui.java:
```diff
@@ -0,0 +1,47 @@
+package zzbot;
+
+import java.util.Scanner;
+
```
another extra line here oops

yooplo reviewed (commented):
Overall, very well done! I could not find much to nitpick on stylewise. A few small updates suggested.

zzzlou committed:
Merge branch 'branch-A-JavaDoc'
necessary

zzzlou committed:
Add line

zzzlou committed:
Merge branch 'branch-A-CodingStandard'
necessary

zzzlou committed:
Add assertions to MainWindow controller

This commit introduces assertions to verify the state of important variables and ensure correct logic flow during runtime. Assertions have been added to check that:

- zzbot is not null when set

These assertions will help catch potential issues early during development.

zzzlou committed:
Merge pull request #1 from zzzlou/branch-A-Assertions

Add assertions to MainWindow controller

zzzlou committed:
Improve code quality by refactoring and adding assertions

Refactored methods to enhance readability and maintainability. Added assertions to ensure null safety for critical variables. Improved method structure and formatting for better code clarity.

zzzlou committed:
Add streams

Implemented Streams feature where necessary

zzzlou committed:
Merge pull request #2 from zzzlou/branch-A-CodeQuality

Improve code quality by refactoring and adding assertions

zzzlou committed:
Merge branch 'master' into branch-A-Streams
necessary

zzzlou committed:
Merge pull request #3 from zzzlou/branch-A-Streams

Branch a streams

zzzlou committed:
Add .yml file

zzzlou committed:
Add help page

zzzlou committed:
Merge branch 'branch-BCD-Extension'
necessary

zzzlou committed:
Improve GUI

zzzlou committed:
Merge branch 'branch-A-BetterGui'
necessary

zzzlou committed:
Add personality

zzzlou committed:
Add more error handling

zzzlou committed:
Add more testing

zzzlou committed:
Add AI Assistance

zzzlou committed:
Add files via upload

zzzlou committed:
Delete docs/Ui.png

zzzlou committed:
Add files via upload

zzzlou committed:
Add accurate name

zzzlou committed:
Merge branch 'master' of github.com:zzzlou/ip

this is to merge local and remote master

zzzlou committed:
Update README.md

zzzlou committed:
Improve code quality

zzzlou committed:
Improve code style

zzzlou committed:
Fix path issues

zzzlou committed:
Update README.md

zzzlou committed:
Update README.md

zzzlou committed:
Update README.md

zzzlou committed:
Update README.md

zzzlou committed:
Update README.md

zzzlou committed:
Update README.md

zzzlou committed:
Update README.md

zzzlou committed:
Fix coding style
